src/node.rs

```diff
@@ -232,6 +232,18 @@ pub enum Type {
     Audio = 6,
 }
 
+impl FromPrimitive for Type {
+    fn from_i64(n: i64) -> Option<Type> {
+        match n {
+            0...6 => Some(unsafe { ::std::mem::transmute(n as u8) }),
```

Comment:
I don't really need the `FromPrimitive` implementation, I just need to be able to convert a `u16` to a `Type`.